### PR TITLE
fix: update fuel system to include center tank transfer and crossfeed capability

### PR DESCRIPTION
**Pull request by donstim**
Much of the current fuel system architecture was carried over from the default Asobo A320NEO. Asobo incorrectly put the A320CEO fuel system architecture into the A320NEO such that fuel burned directly from the center fuel tank. For the A320NEO, fuel is transferred from the center tank to the inner tanks from which it flows to the engines. This transfer takes place under certain conditions (fuel in the center tank and inner tank fuel level less than about 500kg lower than the respective inner tank fuel capacity.) Other changes to make the fuel system more realistic have been made as described in the Summary of Changes section.
## Summary of Changes

1) Crossfeed capability has been added. With the crossfeed valve open and the fuel pumps for one side's inner fuel tank pumps shut down, fuel will be drawn from the other side's inner tank to feed both engines.

2) The Mode Selector overhead pushbutton is now operative. It can be placed in either Automatic mode (the normal mode with no indication showing) or Manual mode (with MAN showing on the lighted pushbutton).

3) The center tank fuel pump pushbuttons on the overhead panel have been changed to control the center tank transfer valves as on the real A320NEO. The A320NEO has jet pumps to transfer fuel from the center tank to the inner tanks. These pumps are activated by suction created by the open valves rather than being traditional pumps that can be directly controlled by the flight crew. NOTE: Until we have a new model, these pushbuttons will continue to be labeled on the overhead panel as center tank pump switches.

4) Center tank fuel will now transfer to the inner tanks rather than being fed directly to the engines. This transfer will take place automatically if both the Mode Sel switch and the center tank valve (labeled as center tank pump) pushbuttons are in the normal AUTO position when the inner tank fuel level reaches a level 500 kg less than the inner fuel tank capacity (i.e., the transfer starts for each inner tank when that tank fuel quantity is less than 5019 kg). The transfer will stop when either there is no center tank fuel, or the respective inner tank is filled. The center tank fuel transfer valves will also close 5 minutes after reaching the center tank reaches the low fuel level (130 kg). Center tank fuel transfer to an inner tank can also be stopped by placing the respective center tank transfer valve pushbutton (labeled as center tank pump) in the OFF position.

5) When the Mode Sel switch is in manual mode, all center tank fuel transfer is controlled manually by the flight crew. Center tank transfer can be started to an inner tank by putting the respective center tank valve pushbutton (labeled as center tank pump) in the ON position.

6) The ECAM fuel page has been updated to show fuel quantity changes, correct valve positioning, and coloration for the added center to inner tank fuel transfers and crossfeeds

7) This update also removes the triggers for autostarting and autostopping engines, which is not supported by the A32NX. It also removes the fuel lines going directly from the engines back to the inner tanks. 

8) Lastly, the trigger to close the outer to inner fuel transfer valves was changed from a below fuel quantity trigger to an above fuel quantity trigger for clarity and simplicity, so that the condition of "true" rather than "false" will close the valves. 

## References

Screenshot (2656)
(Note:  We do not have additional center tanks (ACTs).)
Screenshot (2657)

## Additional context

Discord username (if different from GitHub): donbikes#4084

## Testing instructions
Check that each of the added capabilities described in the "Summary of Changes" section works as described and does not adversely affect any other existing function. Some guidance is provided below. Valve positions and tank fuel quantities can be seen on the ECAM fuel page.

 1) Crossfeed check: With engines running, using the overhead, open the crossfeed valve and shut off the Left tank pumps 1 and 2. Confirm engines remain on and fuel is only being drawn from the right tank. Turn on the left tank fuel pumps and turn off the right tank fuel pumps. Confirm same as previously, except that fuel is now only being drawn from the left tank.

2) Center fuel tank transfer. 
a) Automatic transfer. With the Mode Sel overhead pushbutton in the normal position with neither "Fault" nor "MAN" showing, confirm that any time there is fuel in the center tank and the fuel quantity in any inner tank is less than 5019 kg, the respective jet pump will activate, the transfer valve will open, and fuel will transfer from the center tank to that inner tank. Confirm that when the fuel level in an inner tank where fuel is being transferred from the center tank is full, the fuel transfer to that tank will end. Also, confirm that the center tank transfer valves will close 5 minutes after the center tank reaches 130 kg, and that the transfer valves will close if the center tank valve pushbuttons (labeled as center tank pumps) are put in the OFF position. 
b) Manual transfer. Confirm that with the Mode Sel pushbutton in the MAN position, center tank fuel transfer can be directly controlled by using the center tank valve pushbuttons (labeled as center tank pumps).

3) Confirm that the center tank transfer works correctly with both fueling from a cold and dark start, and also with "refueling" while in flight.

4) Outer tank to inner tank fuel transfer: Confirm that this transfer still works as in the current airplane. If there is fuel in the outer tanks, any time that either inner tank fuel quantity is less than 750 kg, **both** outer to inner tank fuel transfer valves open and fuel is transferred from both outer tanks to **both** inner tanks. When fuel is added to either inner tank to reach a fuel quantity greater than 1422 kg, the outer to inner transfer to that tank will stop (the transfer valve will close). 

4) Confirm that everything else associated with the fuel system still works properly, like APU and engine starts.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Review by Benjozork — APPROVED** [review]

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #7817
Tier of Testing  : 1
Date             : 22/03/2023

Testing Process: 
Crossfeed check: With engines running, using the overhead, open the crossfeed valve and shut off the Left tank pumps 1 and 2. Confirm engines remain on and fuel is only being drawn from the right tank. Turn on the left tank fuel pumps and turn off the right tank fuel pumps. Confirm same as previously, except that fuel is now only being drawn from the left tank. - **OK**, but fuel used is getting updated at the wrong side.

Center fuel tank transfer.
a) Automatic transfer. With the Mode Sel overhead pushbutton in the normal position with neither "Fault" nor "MAN" showing, confirm that any time there is fuel in the center tank and the fuel quantity in any inner tank is less than 5019 kg, the respective jet pump will activate, the transfer valve will open, and fuel will transfer from the center tank to that inner tank. Confirm that when the fuel level in an inner tank where fuel is being transferred from the center tank is full, the fuel transfer to that tank will end. Also, confirm that the center tank transfer valves will close 5 minutes after the center tank reaches 130 kg, and that the transfer valves will close if the center tank valve pushbuttons (labeled as center tank pumps) are put in the OFF position. - **OK**
b) Manual transfer. Confirm that with the Mode Sel pushbutton in the MAN position, center tank fuel transfer can be directly controlled by using the center tank valve pushbuttons (labeled as center tank pumps). - **OK**

Confirm that the center tank transfer works correctly with both fueling from a cold and dark start, and also with "refueling" while in flight. - **OK**

Outer tank to inner tank fuel transfer: Confirm that this transfer still works as in the current airplane. If there is fuel in the outer tanks, any time that either inner tank fuel quantity is less than 750 kg, both outer to inner tank fuel transfer valves open and fuel is transferred from both outer tanks to both inner tanks. When fuel is added to either inner tank to reach a fuel quantity greater than 1422 kg, the outer to inner transfer to that tank will stop (the transfer valve will close). - **OK**

Confirm that everything else associated with the fuel system still works properly, like APU and engine starts. - **OK**

Negatives:       
image
Fuel used is not getting updated correctly for the opposite side engine to the running fuel pumps. Looks like "fuel used" is looking at fuel pumps flow instead of engines fuel flow.

Testing Results: 
Almost passed

Conclusions: 
Crossfeed fuel used not OK.

**Comment by donstim** [comment]
Thanks for the review. This has been fixed and is ready again for further review.

**Comment by 2hwk** [comment]
QA Report

Testing:
- [x] Crossfeed check: With engines running, using the overhead, open the crossfeed valve and shut off the Left tank pumps 1 and 2. 
- [x] Confirm engines remain on and fuel is only being drawn from the right tank. 
- [x] Turn on the left tank fuel pumps and turn off the right tank fuel pumps. Confirm same as previously, except that fuel is now only being drawn from the left tank.
Center fuel tank transfer:

 -[x] Automatic transfer. With the Mode Sel overhead pushbutton in the normal position with neither "Fault" nor "MAN" showing, confirm that any time there is fuel in the center tank and the fuel quantity in any inner tank is less than 5019 kg, the respective jet pump will activate, the transfer valve will open, and fuel will transfer from the center tank to that inner tank. Confirm that when the fuel level in an inner tank where fuel is being transferred from the center tank is full, the fuel transfer to that tank will end. Also, confirm that the center tank transfer valves will close 5 minutes after the center tank reaches 130 kg, and that the transfer valves will close if the center tank valve pushbuttons (labeled as center tank pumps) are put in the OFF position.
- [x] Manual transfer. Confirm that with the Mode Sel pushbutton in the MAN position, center tank fuel transfer can be directly controlled by using the center tank valve pushbuttons (labeled as center tank pumps).
- [x] Confirm that the center tank transfer works correctly with both fueling from a cold and dark start, and also with "refueling" while in flight.
- [x] Outer tank to inner tank fuel transfer: Confirm that this transfer still works as in the current airplane. If there is fuel in the outer tanks, any time that either inner tank fuel quantity is less than 750 kg, both outer to inner tank fuel transfer valves open and fuel is transferred from both outer tanks to both inner tanks. When fuel is added to either inner tank to reach a fuel quantity greater than 1422 kg, the outer to inner transfer to that tank will stop (the transfer valve will close).

 -[x] Confirm that everything else associated with the fuel system still works properly, like APU and engine starts.
 
 Extra:
 
- [x] ECAM fuel page correct symbols + valves status at all stages
- [x] Fuel Used - Properly resets when ENG 1+2 is cut then started again
- [x] ELEC correct at all stages
- [x] HYD correct at all stages
- [x] Spawn states checked - C&D, Runway, Final Approach, Cruise. 

Negatives:       
- None

Testing Results: 
Passed

Conclusions: 
QA Passed, waiting for localazy update for localisation strings before merging

File: fbw-a32nx/src/systems/instruments/src/SD/Pages/Fuel/Fuel.tsx [hunk 1]
```diff
@@ -354,3 +374,97 @@ const Pump = ({ x, y, onBus = 'DC_ESS', pumpNumber, centreTank, tankQuantity }:
...
+    const [transferValveFullyOpen, setTransferValveFullyOpen] = useState(false);
+    const [transferValveFullyClosed, setTransferValveFullyClosed] = useState(false);
+    const [centreTankLowLevel, setCentreTankLowLevel] = useState(false);
+
+    const [pumpClass, setPumpClass] = useState('GreenLine');
+    const [closedClass, setClosedClass] = useState('Hide');
+    const [openClass, setOpenClass] = useState('Hide');
+    const [faultClass, setFaultClass] = useState('Hide');
+    const [transferClass, setTransferClass] = useState('Hide');
+
```
Comment (BlueberryKing) [inline fbw-a32nx/src/systems/instruments/src/SD/Pages/Fuel/Fuel.tsx:404]: Are we sure those need to be stored in state. It seems like they can be computed from the simvars directly and don't need hooks.

Comment (donstim) [reply fbw-a32nx/src/systems/instruments/src/SD/Pages/Fuel/Fuel.tsx:404]: @tracernz or @Taz5150?

Comment (Taz5150) [reply fbw-a32nx/src/systems/instruments/src/SD/Pages/Fuel/Fuel.tsx:404]: Can’t say. I did not work on the Fuel.tsx file for this PR and do not have knowledge on that specific code. Sorry.

**Comment by richardkennedy** [comment]
What are the new controls for the center tank transfer valves? Can't spot them anywhere.

**Comment by tracernz** [comment]
Valves 9 & 10 I think.

**Pull request merged**